### ISSUE TITLE
Differences in visual regression tests

### ISSUE THREAD

**Issue by awocallaghan** [highcharts/highcharts#14030]
#### Expected behaviour
When visual testing charts using services like [Chromatic](https://www.chromaticqa.com/) there should be no visual differences when rendering the same chart.

#### Actual behaviour
There are visual differences between each pipeline run:
https://imgur.com/uFW0Z51
https://imgur.com/3w9orHb

#### Live demo with steps to reproduce
Use a visual diff tool to take screenshots of charts and compare visual differences

#### Product version
Latest version

#### Affected browser(s)
Chrome, IE

**Comment by pawelfus** [comment]
Hi @awocallaghan 

Thank you for contacting us and sorry for the late reply! I did not receive any notification about this issue. I'm not sure if that's a Highcharts issue. If the chart's container width and height are the same (and the configs of course), then the rendered chart should be rendered exactly the same every time, for example: https://jsfiddle.net/BlackLabel/uq8fwthk/1/ - refresh the view multiple times and the results in HTML are always the same.

Some ideas what might be the reason behind this:
- as mentioned above, container's width+height are not the same (e.g. other elements rendering, CSS loading, fonts etc)
- chromatic takes a snapshot of the chart before animation has finished - make sure, for tests, you disabled all animations

I would say that heights for charts are different, thus different result, but this is just a lucky guess.

**Comment by pawelfus** [comment]
Hi @awocallaghan - were you able to find the solution? Or recreate issue in jsFiddle?

**Comment by KacperMadej** [comment]
Closing due to inactivity.

**Comment by andySamKnows** [comment]
I'm experienced this exact issue, did you get a resolution @awocallaghan?

**Comment by zernie** [comment]
Same. The differences are subtle but still present.

**Comment by awocallaghan** [comment]
Quite a few years later and managed to debug the cause of this with the help of Claude Code! 

Here's the findings for why we were having this issue, in case it's helpful for any others suffering from a similar problem:

1. Random SVG ids: Highcharts generates random ids for SVG elements (clip paths, etc.) on every render. Calling `Highcharts.useSerialIds(true)` in our Chromatic environment made the output stable
2. Font-loading race that `document.fonts.ready` didn't fix: We were awaiting `document.fonts.ready` before rendering the charts, but that wasn't enough. The fonts were only requested when the chart was rendered. We fixed this by forcing fonts to load first in a Storybook global loader (eg `document.fonts.load('1em MyFont')`).
3. Animations: We disabled `chart.animation` and `plotOptions.series.animation` within Chromatic so snapshots aren't taken mid-animation.

With all these changes we're getting reliable pixel-identical renders. We've also created a script to render the stories multiple times in a local Storybook, take screenshots with Playwright and diff the screenshots to debug any future non-determinism.
